### DISCUSSION TITLE
I put ViT-B-32__openai models(two onnx files) for smart search in the right folder,how can i tell immich NOT to clear them and NOT to download again?

### DISCUSSION THREAD

**Discussion by lj4o7** [immich-app/immich#11390] [Q&A]
Due to isolated network in China,models can not be download automatically.

**Comment by lj4o7** [comment]
\immich_model-cache\_data\clip\ViT-B-32__openai
folder

**Comment by mertalev** [comment]
Did you only put those two ONNX files? All the JSON files also need to be in the folder.

Alternatively, you can also [use a mirror](https://github.com/immich-app/immich/discussions/6417#discussioncomment-8153066).

**Comment by 1317** [comment]
Somehow [adding a mirror](https://github.com/immich-app/immich/discussions/6417#discussioncomment-8153066) didn't work for me. So I downloaded the models _manually_ via the mirror.

There's an example to download `ViT-B-32__openai` via the mirror.  

**1. Open the terminal of `immich-machine-learning` container**
```shell 
docker exec -it immich_machine_learning /bin/bash
``` 
  or just go to container > Exec if you're using Docker Desktop.
  
**2. Go to the directory where places the CLIP models**
```shell
cd /cache/clip/
``` 
Note that `/cache/clip/` for CLIP and Multilingual CLIP models (Smart Search); `/cache/facial-recognition/` for Facial Recognition models.  

**3. Setup the tools and download the model** _(reference: https://hf-mirror.com/)_
```shell
pip install -U huggingface_hub
export HF_ENDPOINT=https://hf-mirror.com
huggingface-cli download --resume-download immich-app/ViT-B-32__openai --local-dir ViT-B-32__openai
``` 
and use `exit` to quit the container terminal.
Note that the `--local-dir` parameter is followed by `ViT-B-32__openai`, not `immich-app/ViT-B-32__openai `.

**4. (Optional) Check if the model is properly downloaded**
Open the logs of `immich-machine-learning` container:
```shell 
docker logs -f immich_machine_learning
``` 
  or just go to container > Logs if you're using Docker Desktop.
Then start a Smart Search job from the webpage. If everything works fine, you will see logs like below without warnings or errors
````
INFO     Loading textual model                              
             'ViT-B-32__openai' to memory      
````

**5. No need to manually download the models again unless you change to another model.**
